### PR TITLE
Enhancements/ Refactoring of login page and common components

### PR DESCRIPTION
**Pull request by gauravsinhaweb**
Issue 
- #23
Changes

- [x] added test cases for input box
- [x] changed js files to tsx
- [x] updated tests for login page
- [x] feature - show hide password

More refactors suggested by reviewers will be covered in separate ticket:
- #29 
-  #27 
Tests coverage:

Screenshot 2023-10-01 193103

### CODE REVIEW

**Review by iamitprakash — APPROVED** [review]

**Comment by vinit717** [comment]
Tests stats and test for inputBox?

**Comment by gauravsinhaweb** [comment]
> Tests stats and test for inputBox?

Updated

**Comment by sahsisunny** [comment]
image
Can you add the checkbox to show or hide the password?

**Comment by gauravsinhaweb** [comment]
> image
> Can you add the checkbox to show or hide the password?

Yes I'll add,

**Comment by gauravsinhaweb** [comment]
> Tests stats and test for inputBox?

updated

File: package.json
Comment (sahsisunny) [inline package.json:19]: Why?

Comment (gauravsinhaweb) [reply package.json:19]: This tells me exactly where I need to fix the linting, file and line now with the description,
It's optional

File: __tests__/components/InputBox.test.tsx
Comment (Pratiyushkumar) [inline __tests__/components/InputBox.test.tsx:6]: should render what??

File: __tests__/components/InputBox.test.tsx
Comment (Pratiyushkumar) [inline __tests__/components/InputBox.test.tsx:1]: can we not combine both the test cases into one?

File: __tests__/pages/login.test.tsx
Comment (Pratiyushkumar) [inline __tests__/pages/login.test.tsx:22]: can we not combine these two test case into one and test it

File: src/components/InputBox/index.tsx
Comment (Pratiyushkumar) [inline src/components/InputBox/index.tsx:12]: remove types to types file and then import here

File: src/components/InputBox/index.tsx
Comment (Pratiyushkumar) [inline src/components/InputBox/index.tsx:25]: can we take this conditional rendering in  variable and then use it

File: src/components/InputBox/index.tsx
Comment (Pratiyushkumar) [inline src/components/InputBox/index.tsx:44]: let's use this in function and use it here

File: src/pages/dashboard/index.tsx
Comment (Pratiyushkumar) [inline src/pages/dashboard/index.tsx:74]: can we make input and button a reusable component?

**Review by Pratiyushkumar — CHANGES_REQUESTED** [review]

File: __tests__/components/InputBox.test.tsx
Comment (gauravsinhaweb) [reply __tests__/components/InputBox.test.tsx:6]: Inputbox, mentioned in the describe

File: __tests__/components/InputBox.test.tsx
Comment (gauravsinhaweb) [reply __tests__/components/InputBox.test.tsx:1]: ??

File: __tests__/pages/login.test.tsx
Comment (gauravsinhaweb) [reply __tests__/pages/login.test.tsx:22]: No

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx:12]: No need, 
this type is not going to be used by any other component

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx:25]: yes, updated

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx:44]: this is not a function, it is jsx
let me know if you can think of any benefit

File: src/pages/dashboard/index.tsx
Comment (gauravsinhaweb) [reply src/pages/dashboard/index.tsx:74]: @j24m will handle this, had discussion

File: src/components/Footer/index.tsx
Comment (iamitprakash) [inline src/components/Footer/index.tsx:5]: h-[6vh] style on the .bg-gray-200 flex justify-center items-center div will cause the footer to be 6% of the viewport height, even if the viewport is less than 60 pixels high. This means that the footer will overlap the content of the website on small screens.

File: src/components/Layout/index.tsx
Comment (iamitprakash) [inline src/components/Layout/index.tsx:12]: h-container class on the <main> element is not defined, so the browser will ignore it. This means that the height of the <main> element will be determined by the content inside of it, which could cause it to overlap the footer on small screens.

File: __tests__/pages/login.test.tsx
Comment (yesyash) [inline __tests__/pages/login.test.tsx:1]: - Can we setup the `LoginPage` once and reuse it across different assertions ? 
Example: [https://testing-library.com/docs/example-input-event/](https://testing-library.com/docs/example-input-event/)

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: src/components/InputBox/index.tsx
Comment (yesyash) [inline src/components/InputBox/index.tsx:25]: - Instead of having multiple conditional statements here can we do it something like
```js
<input type={showPassword ? 'text' : type} /> 
```

And also set a default value to type in props: 
```js
const { type = "text", name, onChange, placeholder, required, className } = props;
```

File: src/components/InputBox/index.tsx
Comment (yesyash) [inline src/components/InputBox/index.tsx:33]: - Should this check not be based on the input type instead of name ? 
```js
type === "password"
```

File: src/components/Layout/index.tsx
Comment (yesyash) [inline src/components/Layout/index.tsx:14]: - Tip: Instead of wrapping this in a div we can also wrap it in a react fragment. 
```js
return (
    <>
        <Head>
             .....
    </>
)
```

File: src/pages/dashboard/index.tsx
Comment (yesyash) [inline src/pages/dashboard/index.tsx:24]: - Instead of using a % width can we use a definite value here ?

File: src/pages/login/index.tsx
Comment (yesyash) [inline src/pages/login/index.tsx:10]: - Should we move this to a constants file?

File: src/pages/login/index.tsx
Comment (yesyash) [inline src/pages/login/index.tsx:78]: - `href="#"` ?

File: __tests__/components/InputBox.test.tsx
Comment (Pratiyushkumar) [reply __tests__/components/InputBox.test.tsx:6]: sir, dont use shortcuts here... `should render input box looks much better then simply should render

File: __tests__/components/InputBox.test.tsx
Comment (Pratiyushkumar) [reply __tests__/components/InputBox.test.tsx:1]: I mean to say that it why are creating tests for every input box, instead you can combine both tests into one since goal is same.

File: __tests__/pages/login.test.tsx
Comment (Pratiyushkumar) [reply __tests__/pages/login.test.tsx:22]: why sir?

File: src/components/InputBox/index.tsx
Comment (Pratiyushkumar) [reply src/components/InputBox/index.tsx:12]: so why is separate types folder/files made, if this is the case then we don't need separate types folder because maximum time you wont be re-using the same types

File: src/components/InputBox/index.tsx
Comment (Pratiyushkumar) [reply src/components/InputBox/index.tsx:25]: where??? i cant find it

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx:12]: I discussed this with @sahsisunny, and we concluded what I answered

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx:25]: got more comments over the PR, will get pushed in the upcoming commit

File: src/pages/login/index.tsx
Comment (satyam73) [reply src/pages/login/index.tsx:10]: +1

File: src/pages/login/index.tsx
Comment (satyam73) [inline src/pages/login/index.tsx:40]: is this needed?

**Review by satyam73 — CHANGES_REQUESTED** [review]
- At some places we're using function expression for components and at some places we are using normal function, try picking up one thing for consistency. (function will be good here imo)
- We're using layout everywhere, if it is common for all the pages use layout in MyApp only.

File: src/components/Footer/index.tsx
Comment (gauravsinhaweb) [reply src/components/Footer/index.tsx:5]: https://github.com/Real-Dev-Squad/tiny-site-frontend/assets/75125943/77926179-e3dc-42db-b58a-344e55238e0e

File: src/components/Layout/index.tsx
Comment (gauravsinhaweb) [reply src/components/Layout/index.tsx:12]: https://github.com/Real-Dev-Squad/tiny-site-frontend/pull/18#discussion_r1345773798

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx:2]: we can only access src folder using alias not the public folder

Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx:2]: we can only access src folder using alias not the public folder

Comment (satyam73) [reply src/components/InputBox/index.tsx:2]: image

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx]: updated

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx]: done

File: src/components/Layout/index.tsx
Comment (gauravsinhaweb) [reply src/components/Layout/index.tsx]: updated

File: src/pages/dashboard/index.tsx
Comment (gauravsinhaweb) [reply src/pages/dashboard/index.tsx]: updated

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx]: for good readibility

File: src/pages/dashboard/index.js
Comment (gauravsinhaweb) [reply src/pages/dashboard/index.js:1]: this is supposed to be a `.tsx` file

File: __tests__/pages/__snapshots__/login.test.tsx.snap
Comment (sahsisunny) [inline __tests__/pages/__snapshots__/login.test.tsx.snap:1]: Can you remove this type of snapshot file?

File: package.json
Comment (sahsisunny) [reply package.json:19]: modify the already written command

File: src/components/InputBox/index.tsx
Comment (sahsisunny) [reply src/components/InputBox/index.tsx:2]: @satyam73, the public folder is located outside the src directory, which means we cannot use an alias for it.

File: src/pages/dashboard/index.tsx
Comment (sahsisunny) [inline src/pages/dashboard/index.tsx:7]: Can you sort the import if possible?

**Comment by sahsisunny** [comment]
It would be great if you could get approval from Satyam and Pratiyush before I give my approval.

**Comment by sahsisunny** [comment]
Excellent work Gautav, Keep it up :rocket:

File: src/pages/dashboard/index.tsx
Comment (gauravsinhaweb) [reply src/pages/dashboard/index.tsx]: Done

File: src/components/Navbar/index.tsx
Comment (gauravsinhaweb) [reply src/components/Navbar/index.tsx]: Done

File: __tests__/pages/__snapshots__/login.test.tsx.snap
Comment (gauravsinhaweb) [reply __tests__/pages/__snapshots__/login.test.tsx.snap:1]: Updated

**Comment by gauravsinhaweb** [comment]
> It would be great if you could get approval from Satyam and Pratiyush before I give my approval.

@satyam73 @Pratiyushkumar

**Comment by gauravsinhaweb** [comment]
> Excellent work Gaurav, Keep it up 🚀

Thank you sunny

**Review by Pratiyushkumar — APPROVED** [review]

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx]: Updated

File: src/components/InputBox/index.tsx
Comment (gauravsinhaweb) [reply src/components/InputBox/index.tsx]: Updated

File: src/pages/dashboard/index.tsx
Comment (gauravsinhaweb) [reply src/pages/dashboard/index.tsx]: Updated

**Review by vinit717 — APPROVED** [review]

**Review by iamitprakash — CHANGES_REQUESTED** [review]
Add meaningful PR name

**Comment by gauravsinhaweb** [comment]
> Add meaningful PR name

Updated, let me know if you have better suggestion

File: __tests__/components/InputBox.test.tsx
Comment (satyam73) [reply __tests__/components/InputBox.test.tsx:19]: Please fill the ticket with this pointer as one of the todo

File: src/components/InputBox/index.tsx
Comment (satyam73) [reply src/components/InputBox/index.tsx:3]: Yep getting it, it may be setup in different way.
Resolving this comment

File: src/pages/login/index.tsx
Comment (satyam73) [reply src/pages/login/index.tsx]: It would be great attaching a ticket here

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**
